### DISCUSSION TITLE
How to Change the Authenticating Requests Signature of the Object Store S3v2 to S3v4?

### DISCUSSION THREAD

**Discussion by mduerr3cgi** [rook/rook#10075] [Q&A] [answered]
Hello,
I am trying to access the ceph object store via S3v4 api without success. It is working fine if I use S3v2.

Is it somehow possible to change the api via an option?

Thank you in advance,
Marco

**Comment by travisn** [comment]
I thought the v2/v4 had slightly different flags to work with rgw, but I can't find my notes on it now. 
@thotz @alimaredia Do you know?

**Reply by thotz** [reply]
There are certain apis like creating topics fails with v4 see https://tracker.ceph.com/issues/50039 for details, but most of APIs work with v4 IMO. AFAIR there has no server-side option to enable v4 authentication explicitly. Can you share the s3 API which is performing the s3 client which is using and logs of rgw server with higher log level

**Reply by mduerr3cgi** [reply]
Thank you for the response.

In the picture below you can see two approaches to access the rgw with the minio client (mcli). The green marked one is successfull, the red one not. The ERRORHANDLER throws error number: 2027

My actual goal is to use the restic backup client, but over there the error messages are exactly the same.

![s3-api-edit](https://user-images.githubusercontent.com/103638698/164191478-9195b019-9f8c-49bb-9e30-846670f42226.png)

**Reply by cbodley** [reply]
> AFAIR there has no server-side option to enable v4 authentication explicitly

that's right, radosgw supports both v2 and v4 signatures. the client chooses which method to use when signing its requests

it looks like there's a v4 interop issue between mcli and radosgw here, so we'll need more information to track it down. can you enable debug logging (debug_rgw = 20) in radosgw to capture the details of this request? that will show all of the request headers that go into this signature calculation

edit: debug output from `mcli ls test --debug` could be helpful too

**Reply by cbodley** [reply]
> mcli ls test

does `mcli ls` work without the bucket name? if so, this is probably because the client is sending [vhost-style requests](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html), and radosgw needs extra configuration ([rgw_dns_name](https://docs.ceph.com/en/latest/radosgw/config-ref/#confval-rgw_dns_name)) to support that. i don't know whether rook configures that for you

**Reply by thotz** [reply]
we can config options either in rook-ceph configmap. or u can run `ceph config set` from the toolbox pod. Let me know if you need help in setting the options(log level or rgw_dns_name)

**Reply by mduerr3cgi** [reply]
I have changed the log level of the rgw by executing the following command inside of the rook ceph toolbox pod: `ceph config set global debug_rgw 20`  

**RGW output of the mcli request to the v4 api:**  

debug 2022-04-21T10:35:40.233+0000 7f83b62e2700  1 ====== starting new request req=0x7f844557d620 =====                                                     debug 2022-04-21T10:35:40.233+0000 7f83b62e2700  2 req 17385225719812524942 0.000000000s initializing for trans_id = tx00000f144baae69690b8e-006261337c-2cc567-my-store                                                                                                                                                 debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s rgw api priority: s3=8 s3website=7                                 debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s host=192.168.0.10                                                  debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 20 req 17385225719812524942 0.000000000s subdomain= domain= in_hosted_domain=0 in_hosted_domain_s3website=0 debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 20 req 17385225719812524942 0.000000000s final domain/bucket subdomain= domain= in_hosted_domain=0 in_hosted_domain_s3website=0 s->info.domain= s->info.request_uri=/                                                                                                   debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s meta>> HTTP_X_AMZ_CONTENT_SHA256                                   debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s meta>> HTTP_X_AMZ_DATE                                             debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s x>> x-amz-content-sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855                                                                                                                                      debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s x>> x-amz-date:20220421T103540Z                                    debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 20 req 17385225719812524942 0.000000000s get_handler handler=26RGWHandler_REST_Service_S3                   debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s handler=26RGWHandler_REST_Service_S3                               debug 2022-04-21T10:35:40.233+0000 7f83b62e2700  2 req 17385225719812524942 0.000000000s getting op 0                                                       debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s s3:list_buckets scheduling with throttler client=3 cost=1          debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s s3:list_buckets op=26RGWListBuckets_ObjStore_S3                    debug 2022-04-21T10:35:40.233+0000 7f83b62e2700  2 req 17385225719812524942 0.000000000s s3:list_buckets verifying requester                                debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 20 req 17385225719812524942 0.000000000s s3:list_buckets rgw::auth::StrategyRegistry::s3_main_strategy_t: trying rgw::auth::s3::AWSAuthStrategy
debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 20 req 17385225719812524942 0.000000000s s3:list_buckets rgw::auth::s3::AWSAuthStrategy: trying rgw::auth::s3::S3AnonymousEngine
debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 20 req 17385225719812524942 0.000000000s s3:list_buckets rgw::auth::s3::S3AnonymousEngine denied with reason=-1
debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 20 req 17385225719812524942 0.000000000s s3:list_buckets rgw::auth::s3::AWSAuthStrategy: trying rgw::auth::s3::LocalEngine
debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s v4 signature format = bb0082988d6edca0f80b05134da1bcb3e3190bef68b70dc6d07f05efcca1d3c3                                                                                                                                         debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s v4 credential format = MVW69ZKCB9OCBDA38XQL/20220421/us-east-1/s3/aws4_request                                                                                                                                                 debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s access key id = MVW69ZKCB9OCBDA38XQL
debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s credential scope = 20220421/us-east-1/s3/aws4_request
debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s canonical headers format = host:192.168.0.10
x-amz-content-sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date:20220421T103540Z
debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s payload request hash = e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s canonical request = GET
/

host:192.168.0.10
x-amz-content-sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date:20220421T103540Z

host;x-amz-content-sha256;x-amz-date
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s canonical request hash = 2db75624f81e004f5440acb69a2d27540a4693a254af6d6057cd32e4e52837e3
debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s string to sign = AWS4-HMAC-SHA256
20220421T103540Z
20220421/us-east-1/s3/aws4_request
2db75624f81e004f5440acb69a2d27540a4693a254af6d6057cd32e4e52837e3
debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 20 req 17385225719812524942 0.000000000s s3:list_buckets get_system_obj_state: rctx=0x7f844557bae8 obj=my-store.rgw.meta:users.keys:MVW69ZKCB9OCBDA38XQL state=0x558eb4db3720 s->prefetch_data=0
debug 2022-04-21T10:35:40.233+0000 7f83b62e2700 10 req 17385225719812524942 0.000000000s s3:list_buckets cache get: name=my-store.rgw.meta+users.keys+MVW69ZKCB9OCBDA38XQL : expiry miss
debug 2022-04-21T10:35:40.234+0000 7f83161a2700 10 req 17385225719812524942 0.000999955s s3:list_buckets cache put: name=my-store.rgw.meta+users.keys+MVW69ZKCB9OCBDA38XQL info.flags=0x6
debug 2022-04-21T10:35:40.234+0000 7f83161a2700 10 req 17385225719812524942 0.000999955s s3:list_buckets adding my-store.rgw.meta+users.keys+MVW69ZKCB9OCBDA38XQL to cache LRU end
debug 2022-04-21T10:35:40.234+0000 7f83161a2700 20 req 17385225719812524942 0.000999955s s3:list_buckets get_system_obj_state: s->obj_tag was set empty
debug 2022-04-21T10:35:40.234+0000 7f83161a2700 10 req 17385225719812524942 0.000999955s s3:list_buckets cache get: name=my-store.rgw.meta+users.keys+MVW69ZKCB9OCBDA38XQL : type miss (requested=0x1, cached=0x6)
debug 2022-04-21T10:35:40.234+0000 7f83161a2700 20 req 17385225719812524942 0.000999955s s3:list_buckets rados->read ofs=0 len=0
debug 2022-04-21T10:35:40.237+0000 7f8376262700 20 req 17385225719812524942 0.003999822s s3:list_buckets rados_obj.operate() r=0 bl.length=9
debug 2022-04-21T10:35:40.237+0000 7f8376262700 10 req 17385225719812524942 0.003999822s s3:list_buckets cache put: name=my-store.rgw.meta+users.keys+MVW69ZKCB9OCBDA38XQL info.flags=0x1
debug 2022-04-21T10:35:40.237+0000 7f8376262700 10 req 17385225719812524942 0.003999822s s3:list_buckets moving my-store.rgw.meta+users.keys+MVW69ZKCB9OCBDA38XQL to cache LRU end
debug 2022-04-21T10:35:40.237+0000 7f8376262700 20 req 17385225719812524942 0.003999822s s3:list_buckets get_system_obj_state: rctx=0x7f844557bae8 obj=my-s$ore.rgw.meta:users.uid:duerr state=0x558eb48f9de0 s->prefetch_data=0
debug 2022-04-21T10:35:40.237+0000 7f8376262700 10 req 17385225719812524942 0.003999822s s3:list_buckets cache get: name=my-store.rgw.meta+users.uid+duerr : expiry miss
debug 2022-04-21T10:35:40.239+0000 7f83a8ac7700 10 req 17385225719812524942 0.005999732s s3:list_buckets cache put: name=my-store.rgw.meta+users.uid+duerr info.flags=0x16
debug 2022-04-21T10:35:40.239+0000 7f83a8ac7700 10 req 17385225719812524942 0.005999732s s3:list_buckets adding my-store.rgw.meta+users.uid+duerr to cache LRU end
debug 2022-04-21T10:35:40.239+0000 7f83a8ac7700 10 req 17385225719812524942 0.005999732s s3:list_buckets updating xattr: name=ceph.objclass.version bl.length()=42
debug 2022-04-21T10:35:40.239+0000 7f83a8ac7700 20 req 17385225719812524942 0.005999732s s3:list_buckets get_system_obj_state: s->obj_tag was set empty
debug 2022-04-21T10:35:40.239+0000 7f83a8ac7700 10 req 17385225719812524942 0.005999732s s3:list_buckets cache get: name=my-store.rgw.meta+users.uid+duerr : type miss (requested=0x11, cached=0x16)
debug 2022-04-21T10:35:40.239+0000 7f83a8ac7700 20 req 17385225719812524942 0.005999732s s3:list_buckets rados->read ofs=0 len=0
debug 2022-04-21T10:35:40.241+0000 7f83c3afd700 20 req 17385225719812524942 0.007999643s s3:list_buckets rados_obj.operate() r=0 bl.length=377
debug 2022-04-21T10:35:40.241+0000 7f83c3afd700 10 req 17385225719812524942 0.007999643s s3:list_buckets cache put: name=my-store.rgw.meta+users.uid+duerr info.flags=0x11
debug 2022-04-21T10:35:40.241+0000 7f83c3afd700 10 req 17385225719812524942 0.007999643s s3:list_buckets moving my-store.rgw.meta+users.uid+duerr to cache LRU end
debug 2022-04-21T10:35:40.241+0000 7f83c3afd700 10 req 17385225719812524942 0.007999643s s3:list_buckets chain_cache_entry: cache_locator=my-store.rgw.meta+users.uid+duerr
debug 2022-04-21T10:35:40.241+0000 7f83c3afd700 10 req 17385225719812524942 0.007999643s date_k    = 3e995966657de6dc0111461e46d7f2cedccad373dc3c338bd75109037d18b230
debug 2022-04-21T10:35:40.241+0000 7f83c3afd700 10 req 17385225719812524942 0.007999643s region_k  = 4c2725bd1251e27563cb3de076f201444f0c8b45854d8f064be81b0a1e8adc3f
debug 2022-04-21T10:35:40.241+0000 7f83c3afd700 10 req 17385225719812524942 0.007999643s service_k = d3fd7987437c9eb8f4161574e864e485271e6a7387866e559071ea9d1884c6f0
debug 2022-04-21T10:35:40.241+0000 7f83c3afd700 10 req 17385225719812524942 0.007999643s signing_k = f28332213c625d486cd7bd0f9a2483afba1c45080dd962a4025091d685b47551
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s generated signature = 8c2d6631ec478ed9110d8d61deb14c1a02dff9ebbfb5e61b349eb429c03b9001
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 15 req 17385225719812524942 0.008999599s s3:list_buckets string_to_sign=AWS4-HMAC-SHA256
20220421T103540Z
20220421/us-east-1/s3/aws4_request
2db75624f81e004f5440acb69a2d27540a4693a254af6d6057cd32e4e52837e3
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 15 req 17385225719812524942 0.008999599s s3:list_buckets server signature=8c2d6631ec478ed9110d8d61deb14c1a02dff9ebbfb5e61b349eb429c03b9001
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 15 req 17385225719812524942 0.008999599s s3:list_buckets client signature=bb0082988d6edca0f80b05134da1bcb3e3190bef68b70dc6d07f05efcca1d3c3
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 15 req 17385225719812524942 0.008999599s s3:list_buckets compare=42
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 20 req 17385225719812524942 0.008999599s s3:list_buckets rgw::auth::s3::LocalEngine denied with reason=-2027debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 20 req 17385225719812524942 0.008999599s s3:list_buckets rgw::auth::s3::AWSAuthStrategy denied with reason=-2027
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 20 req 17385225719812524942 0.008999599s s3:list_buckets rgw::auth::StrategyRegistry::s3_main_strategy_t: trying rgw::auth::s3::AWSAuthStrategy
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 20 req 17385225719812524942 0.008999599s s3:list_buckets rgw::auth::s3::AWSAuthStrategy: trying rgw::auth::s3::LocalEngine
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s v4 signature format = bb0082988d6edca0f80b05134da1bcb3e3190bef68b70dc6d07f05efcca1d3c3
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s v4 credential format = MVW69ZKCB9OCBDA38XQL/20220421/us-east-1/s3/aws4_request
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s access key id = MVW69ZKCB9OCBDA38XQL
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s credential scope = 20220421/us-east-1/s3/aws4_request
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s canonical headers format = host:192.168.0.10
x-amz-content-sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date:20220421T103540Z

debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s payload request hash = e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s canonical request = GET
/

host:192.168.0.10
x-amz-content-sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date:20220421T103540Z

host;x-amz-content-sha256;x-amz-date
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s canonical request hash = 2db75624f81e004f5440acb69a2d27540a4693a254af6d6057cd32e4e52837e3
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s string to sign = AWS4-HMAC-SHA256
20220421T103540Z
20220421/us-east-1/s3/aws4_request
2db75624f81e004f5440acb69a2d27540a4693a254af6d6057cd32e4e52837e3
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s date_k    = 3e995966657de6dc0111461e46d7f2cedccad373dc3c338bd75109037d18b230
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s region_k  = 4c2725bd1251e27563cb3de076f201444f0c8b45854d8f064be81b0a1e8adc3f
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s service_k = d3fd7987437c9eb8f4161574e864e485271e6a7387866e559071ea9d1884c6f0
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s signing_k = f28332213c625d486cd7bd0f9a2483afba1c45080dd962a4025091d685b47551
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 req 17385225719812524942 0.008999599s generated signature = 8c2d6631ec478ed9110d8d61deb14c1a02dff9ebbfb5e61b349eb429c03b9001
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 15 req 17385225719812524942 0.008999599s s3:list_buckets string_to_sign=AWS4-HMAC-SHA256
20220421T103540Z
20220421/us-east-1/s3/aws4_request
2db75624f81e004f5440acb69a2d27540a4693a254af6d6057cd32e4e52837e3
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 15 req 17385225719812524942 0.008999599s s3:list_buckets server signature=8c2d6631ec478ed9110d8d61deb14c1a02dff9ebbfb5e61b349eb429c03b9001
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 15 req 17385225719812524942 0.008999599s s3:list_buckets client signature=bb0082988d6edca0f80b05134da1bcb3e3190bef68b70dc6d07f05efcca1d3c3
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 15 req 17385225719812524942 0.008999599s s3:list_buckets compare=42
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 20 req 17385225719812524942 0.008999599s s3:list_buckets rgw::auth::s3::LocalEngine denied with reason=-2027debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 20 req 17385225719812524942 0.008999599s s3:list_buckets rgw::auth::s3::AWSAuthStrategy denied with reason=-2027
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700  5 req 17385225719812524942 0.008999599s s3:list_buckets Failed the auth strategy, reason=-2027
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700 10 failed to authorize request
debug 2022-04-21T10:35:40.242+0000 7f83c3afd700  1 req 17385225719812524942 0.008999599s op->ERRORHANDLER: err_no=-2027 new_err_no=-2027
debug 2022-04-21T10:35:40.243+0000 7f831319c700  2 req 17385225719812524942 0.009999554s s3:list_buckets op status=0
debug 2022-04-21T10:35:40.244+0000 7f831319c700  2 req 17385225719812524942 0.010999509s s3:list_buckets http status=403
debug 2022-04-21T10:35:40.244+0000 7f831319c700  1 ====== req done req=0x7f844557d620 op status=0 http_status=403 latency=0.010999509s ======
debug 2022-04-21T10:35:40.244+0000 7f831319c700  1 beast: 0x7f844557d620: 10.233.90.184 - - [21/Apr/2022:10:35:40.231 +0000] "GET / HTTP/1.1" 403 201 - "MinIO (linux; amd64) minio-go/v7.0.24 mcli/RELEASE.2022-04-16T21-11-21Z" - latency=0.010999509s

**mcli verbose output:**  

mcli: <DEBUG> GET / HTTP/1.1
Host: 10.162.0.100
User-Agent: MinIO (linux; amd64) minio-go/v7.0.24 mcli/RELEASE.2022-04-16T21-11-21Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20220421/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20220421T103540Z
Accept-Encoding: gzip

mcli: <DEBUG> HTTP/1.1 403 Forbidden
Content-Length: 201
Accept-Ranges: bytes
Connection: keep-alive
Content-Type: application/xml
Date: Thu, 21 Apr 2022 10:35:40 GMT
Server: nginx
X-Amz-Request-Id: tx00000f144baae69690b8e-006261337c-2cc567-my-store

<?xml version="1.0" encoding="UTF-8"?><Error><Code>SignatureDoesNotMatch</Code><RequestId>tx00000f144baae69690b8e-006261337c-2cc567-my-store</RequestId><HostId>2cc567-my-store-my-store</HostId></Error>mcli: <DEBUG> Response Time:  62.1652ms

mcli: <ERROR> Unable to list folder. The request signature we calculated does not match the signature you provided. Check your key and signing method.
 (1) ls.go:248 cmd.doList(..) Tags: [http://10.162.0.100/]
 (0) client-s3.go:2091 cmd.(*S3Client).listInRoutine(..)
 Commit:0530f5a47110c7a9325212d6093a214be0b8fd60 | Release-Tag:RELEASE.2022-04-16T21-11-21Z | Host:DE-L084160 | OS:linux | Arch:amd64 | Lang:go1.18.1 | Mem:11 MB/34 MB | Heap:11 MB/24 MB

**Comment by mduerr3cgi** [comment, marked as answer]
Problem Solved: The reverse proxy configuration was wrong, hence the server signature was wrong, too. Including the correct headers on the reverse proxy configuration solved the problem.

**Reply by thotz** [reply]
@mduerr3cgi : I am guessing u are referring to minio cli configurations, can you please share those so it will help for other users
